scripts/assets/file_content_validation_checks.textproto

```diff
@@ -371,4 +373,25 @@ file_content_checks {
   prohibited_content_regex: "@color/(?!color_def_).+"
   failure_message: "Only colors from color_defs.xml may be used in color_palette.xml."
 }
-
+file_content_checks {
+  file_path_regex: ".+?ActivityTest.kt"
+  failure_message: "Screen Name test not added."
+  exempted_file_patterns: "app/src/sharedTest/java/org/oppia/android/app/testing.+?\\.kt"
+  exempted_file_patterns: "app/src/test/java/org/oppia/android/app/testing/activity.+?\\.kt"
+  required_content_regex: "testActivity_createIntent_verifyScreenNameInIntent\\("
+}
+file_content_checks {
+  file_path_regex: ".+?Activity.kt"
+  failure_message: "Screen Name not added."
+  exempted_file_name: "app/src/main/java/org/oppia/android/app/activity/InjectableAppCompatActivity.kt"
+  exempted_file_patterns: "app/src/main/java/org/oppia/android/app/testing.+?\\.kt"
+  exempted_file_patterns: "app/src/main/java/org/oppia/android/app/devoptions/testing.+?\\.kt"
+  exempted_file_patterns: "app/src/main/java/org/oppia/android/app/devoptions/vieweventlogs/testing.+?\\.kt"
+  exempted_file_patterns: "app/src/main/java/org/oppia/android/app/devoptions/markchapterscompleted/testing.+?\\.kt"
+  exempted_file_patterns: "app/src/main/java/org/oppia/android/app/devoptions/marktopicscompleted/testing.+?\\.kt"
+  exempted_file_patterns: "app/src/main/java/org/oppia/android/app/devoptions/forcenetworktype/testing.+?\\.kt"
+  exempted_file_patterns: "app/src/main/java/org/oppia/android/app/devoptions/markstoriescompleted/testing.+?\\.kt"
+  exempted_file_patterns: "app/src/main/java/org/oppia/android/app/player/state/testing.+?\\.kt"
+  exempted_file_patterns: "testing.+?\\.kt"
+  required_content_regex: "org\\.oppia\\.android\\.app\\.model\\.ScreenName"
```

Comment:
Could we perhaps match against passing a ScreenName into the needed function? Both can technically be "faked" but this one I'm concerned will have more false positives.

Reply:
I'm not sure if we can test for the exact ScreenName value (since all enum values are different and imported directly) but I've modified the required_content_regex to match for `decorateWithScreenName(`. This ensures that the screen name is added since the app won't build without a screen name being checked into this function. Does that work?

Reply:
Yeah, this seems more robust than the previous pattern. Thanks!